core/CronArchive.php

```diff
@@ -1815,9 +1858,16 @@ private function getUrlsWithSegment($idSite, $period, $date)
                 continue;
             }
 
+            list($visits, $isThereArchive) = $this->isThereAValidArchiveForPeriod($idSite, $period, $segment);
```

Comment:
@diosmosis I wonder if we should move this into line `1907` after 

```php

                if ($self->isAlreadyArchivingSegment($urlWithSegment, $idSite, $period, $segment)) {
                    return Request::ABORT;
                }
```
and then make it like
```php
list($visits, $isThereArchive) = $this->isThereAValidArchiveForPeriod($idSite, $period, $segment);
if ($isThereArchive) {
   return Request::ABORT;
}
```

Reason being there might be 20 segments for example, each segment could take an hour to archive, and by the time it actually executes a segment it is invalidated. So we'd need to check it really only shortly before executing a request.

Reply:
👍